src/main/java/org/tailormap/api/solr/SolrHelper.java

```diff
@@ -170,30 +168,37 @@ public SearchIndex addFeatureTypeIndex(
             propertyName -> {
               Object value = feature.getAttribute(propertyName);
               if (value != null) {
-                if (value instanceof Geometry) {
+                if (value instanceof Geometry
+                    && propertyName.equals(tmFeatureType.getDefaultGeometryAttribute())) {
                   doc.setGeometry(GeometryProcessor.processGeometry(value, true, true, null));
                 } else {
-                  // when display and/or search fields are configured, add the value to the search
-                  // and/or display field otherwise add the value to the search and display field
                   if (searchFields.contains(propertyName)) {
                     searchValues.add(value.toString());
                   }
-                  if (hasDisplayFields) {
-                    if (displayFields.contains(propertyName)) {
-                      displayValues.add(value.toString());
-                    }
+                  if (displayFields.contains(propertyName)) {
+                    displayValues.add(value.toString());
                   }
                 }
               }
             });
-        doc.setSearchFields(searchValues.toArray(new String[searchFields.size() + 2]));
-        doc.setDisplayFields(displayValues.toArray(new String[0]));
-        docsBatch.add(doc);
+        if (searchValues.isEmpty() || displayValues.isEmpty()) {
+          // this is a record/document that can either not be found or not be displayed
+          logger.warn(
```

Comment:
```suggestion
          logger.trace(
```
A warning with total number is logged at the end, don't think we need a log line for every feature at the default warning level.